File: fabric-audit-agent/core/routing.js
import { domainOf } from './key-utils.js';

export const DEFAULT_ROUTES = {
  security: 'security-team',
  cost: 'finops',
  capacity: 'powerbi-team',
  model: 'powerbi-team',
  report: 'powerbi-team',
  pipeline: 'powerbi-team',
  lineage: 'powerbi-team',
  meta: 'powerbi-team',
};

/**
 * Group finding keys by destination owner based on their domain. Pure.
 * @param {object[]} findings  findings with .key
 * @param {Record<string,string>} [routes]  domain -> destination
 * @returns {Record<string, string[]>}  destination -> finding keys
 */
export function routeFindings(findings = [], routes = DEFAULT_ROUTES) {
  const routed = {};
  for (const f of findings) {
    const dest = routes[domainOf(f.key)] ?? 'unrouted';
    (routed[dest] ??= []).push(f.key);
  }
  return routed;
}

File: fabric-audit-agent-app/fabric_audit_agent/routing.py
"""Route finding keys to owners by domain. Port of ``core/routing.js``. Pure."""
from .key_utils import domain_of

DEFAULT_ROUTES = {
    "security": "security-team", "cost": "finops", "capacity": "powerbi-team",
    "model": "powerbi-team", "report": "powerbi-team", "pipeline": "powerbi-team",
    "lineage": "powerbi-team", "meta": "powerbi-team",
}


def route_findings(findings=None, routes=None):
    findings = findings or []
    routes = routes if routes is not None else DEFAULT_ROUTES
    routed = {}
    for f in findings:
        dest = routes.get(domain_of(f.get("key")), "unrouted")
        routed.setdefault(dest, []).append(f.get("key"))
    return routed
